app/src/main/java/openfoodfacts/github/scrachx/openfood/analytics/AnalyticsEvent.kt

```diff
@@ -0,0 +1,111 @@
+package openfoodfacts.github.scrachx.openfood.analytics
+
+sealed class AnalyticsEvent {
+
+    abstract val category: String
+    abstract val action: String
+    abstract val name: String?
+    abstract val value: Float?
+
+    data class ScannedBarcode(override val name: String) : AnalyticsEvent() {
```

Comment:
Instead of overriding each time, let's create a constructon with the values and then initialize every class by calling super.
Like 
```
data class ScannedBarcode(name: String) : AnalysicsEvent(name, "scanner", "scanned", null);
```